code/modules/antagonists/changeling/powers/lesserform.dm

```diff
@@ -5,49 +5,62 @@
 	button_icon_state = "lesser_form"
 	chemical_cost = 5
 	dna_cost = 1
-	req_human = TRUE
 
 /datum/action/changeling/lesserform/Grant(mob/granted_to)
 	. = ..()
-	RegisterSignal(granted_to, COMSIG_HUMAN_MONKEYIZE, PROC_REF(swap_powers))
+	if (!owner)
+		return
+	RegisterSignals(granted_to, list(COMSIG_HUMAN_MONKEYIZE, COMSIG_MONKEY_HUMANIZE), PROC_REF(changed_form))
 
 /datum/action/changeling/lesserform/Remove(mob/remove_from)
-	UnregisterSignal(remove_from, COMSIG_HUMAN_MONKEYIZE)
+	UnregisterSignal(remove_from, list(COMSIG_HUMAN_MONKEYIZE, COMSIG_MONKEY_HUMANIZE))
 	return ..()
 
 //Transform into a monkey.
 /datum/action/changeling/lesserform/sting_action(mob/living/carbon/human/user)
 	if(!user || user.notransform)
 		return FALSE
 	..()
+	return ismonkey(user) ? unmonkey(user) : become_monkey(user)
+
+/// Stop being a monkey
+/datum/action/changeling/lesserform/proc/unmonkey(mob/living/carbon/human/user)
+	if(user.movement_type & VENTCRAWLING)
+		user.balloon_alert(user, "exit pipes!")
```

Comment:
leaving the pipes is already quite a visual/intuitive thing so i don't think this balloon alert makes sense/is useful from a UX standpoint, probably best to do without it

Reply:
This is an instruction (you can't do this because you are in some pipes) rather than a notification but the fact that you might have confused it for something else tells me that I went too far in trying to keep it brief.

Reply:
oh i misunderstood that entire proc. new balloon alert definitely better though